File: grida-domains/domain/src/main/java/org/grida/domain/diary/DiaryImageRepository.java
package org.grida.domain.diary;

import org.grida.domain.core.ImageDetail;

import java.time.LocalDateTime;
import java.util.List;

public interface DiaryImageRepository {
    long save(long diaryId, ImageDetail imageDetail, LocalDateTime lastActionAt);
    DiaryImage findActivateImageByDiaryId(long diaryId);
    List<DiaryImage> findAllByDiaryId(long diaryId);
    void deactivateImage(long diaryId);
    void deleteAllByDiaryId(long diaryId);
}

File: grida-core/core-domain/src/main/kotlin/org/grida/domain/diaryimage/DiaryImageRepository.kt
package org.grida.domain.diaryimage

import org.grida.domain.diary.Diary
import org.grida.domain.image.ImageStatus
import org.grida.domain.user.User

interface DiaryImageRepository {
    fun save(diaryImage: DiaryImage, diary: Diary, user: User): Long

    fun findById(id: Long): DiaryImage

    fun countByDiaryId(diaryId: Long): Long

    fun existsByDiaryIdAndStatus(diaryId: Long, status: ImageStatus): Boolean

    fun updateStatus(diaryImageId: Long, status: ImageStatus): Long
}